### PR TITLE
Airport Challenge - Webb

### PR DESCRIPTION
**Pull request by christopherxwebb**
The user stories are complete but there are still a few edge cases that need ironing out (planes can currently takeoff an infinite amount of times, etc.).

The test coverage is >95% but there are certain tests (weather tests) that remain incomplete.

### CODE REVIEW

File: spec/airport_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    airport.takeoff(plane)
+    expect(airport).to respond_to :takeoff
+  end
+
+  #it "raises error if pilot attempts to land and airport is full" do
+    #@planes == 1
+   # expect { Airport.new.land Plane.new }.to raise_error
+  #end
+
+  it "raise an error when you try to land a plane in a full airport" do
```
Comment (chris-clement) [inline spec/airport_spec.rb:34]: Well written test. I can see exactly what it is setting out to achieve.

File: lib/weather.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require_relative 'plane'
+require_relative 'airport'
```
Comment (chris-clement) [inline lib/weather.rb:2]: Is this require_relative necessary? Is anything from plane or airport being used in this weather file. I think you maybe only need to require_relative in the airport program.

File: lib/weather.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+class Weather
+
+def stormy?
```
Comment (chris-clement) [inline lib/weather.rb:6]: I'd suggest adding indentation as it is within your Weather class.

File: lib/weather.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+require_relative 'plane'
+require_relative 'airport'
+
+class Weather
```
Comment (chris-clement) [inline lib/weather.rb:4]: I like that you made Weather a separate class.

File: lib/airport.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    @capacity = desired_capacity
+    @weather = weather
+  end
+
+  def land(plane)
+    if @planes.count == 0 && @weather.stormy? == "false"
+        @planes.push(plane)
+    elsif @planes.count == DEFAULT_CAPACITY
+        fail "Airport full"
+    else fail "Weather is too stormy"
```
Comment (chris-clement) [inline lib/airport.rb:22]: Should 'fail "Weather is too stormy"' be under the else statement? This reads more like a if or elsif condition

File: spec/airport_spec.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    airport = Airport.new
+    Airport::DEFAULT_CAPACITY.times { airport.land Plane.new }
```
Comment (chris-clement) [inline spec/airport_spec.rb:36]: good use of a constant here rather than a magic number.

**Review by chris-clement — COMMENTED** [review]
Nice work! Lots of good code here, just a few suggestions which I have added comments for.

File: lib/weather.rb (hunk 2 above)
Comment (christopherxwebb) [reply lib/weather.rb:2]: A really good point. I think I did something which I'll try to avoid next time: i.e. just blanket put in 'require_relative' with the assumption that in the future these files will be interacting with each other. But they never do so is actually superfluous. Thanks!

File: lib/airport.rb (hunk 5 above)
Comment (christopherxwebb) [reply lib/airport.rb:22]: Yes, good point.

**Comment by christopherxwebb** [comment]
Thanks, Chris! Really helpful comments, which I'll be sure to incorporate on my next commit.
